RIOT-OS/RIOT
# sys/random: add SHA256 mode to SHA1PRNG & tests

### Contribution description
This PR adds a SHA256PRNG pseudo random number generator using the existing SHA1PRNG implementation. It additionally extends the documentation of these generators, adds two separate unit tests and fixes two issues in the init function (reset buffer position value and use digest instead of state for seeding )

### Testing procedure

- `make -C tests/unittests/ tests-prng_sha1prng  all term` -> test success
- `make -C tests/unittests/ tests-prng_sha256prng  all term` -> test success
- `USEMODULE=prng_sha1prng make -C tests/rng all term` (see README for further information)
- `USEMODULE=prng_sha256prng make -C tests/rng all term` (see README for further information)
